### ISSUE TITLE
[BUG] MONAI ConfigParser reads config files in a list

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/monai-deploy-app-sdk#364]
**Describe the bug**

Hi @MMelQin @ericspod ,

I tried to understand the bundle parsing logic in MONAI deploy app, and I feel this part may be not correct:
https://github.com/Project-MONAI/monai-deploy-app-sdk/blob/main/monai/deploy/operators/monai_bundle_inference_operator.py#L92
`ConfigParser` should read all the config files together and merge them, but current implementation only keeps the last one.
`ConfigParser` logic:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/config_parser.py#L294

**Comment by MMelQin** [comment]
> **Describe the bug**
> 
> Hi @MMelQin @ericspod ,
> 
> I tried to understand the bundle parsing logic in MONAI deploy app, and I feel this part may be not correct: https://github.com/Project-MONAI/monai-deploy-app-sdk/blob/main/monai/deploy/operators/monai_bundle_inference_operator.py#L92 `ConfigParser` should read all the config files together and merge them, but here is logic will only keep the last one. `ConfigParser` logic: https://github.com/Project-MONAI/MONAI/blob/dev/monai/bundle/config_parser.py#L294

Thanks @Nic-Ma for the comments and the issue. I believe the existing implementation was due to the fact that for App SDK we specifically only need the inference.json, with only the input/output tensor names/types are needed from metadata.json, and the data is consumed by separate sections of the app code.

Anyway, I will look into this and test out the code with the "read all config files" logic.

**Comment by MMelQin** [comment]
@Nic-Ma I've created a PR #378 to implement the enhancement to address this issue.

For runtime performance reasons, only files for the specified configs are extracted to temp dir. Tried to use mmap file but the underlying config parser function does not support it.

One thing still not clear to me: should the metadata.json be read together with other config files, e.g. inference.json? Right now, the metadata.json is only read using the read_meta function.
